### PR TITLE
ContainerInfra V1: add Cluster resource

### PR DESCRIPTION
**Pull request by ozerovandrei**
Add the new "openstack_containerinfra_cluster_v1" resource with basic acceptance test.

For #402 

References: 
[Cluster statuses](https://github.com/openstack/magnum/blob/stable/rocky/magnum/objects/fields.py#L18)
[Cluster creation timeout](https://github.com/openstack/magnum/blob/stable/rocky/magnum/conf/cluster_heat.py#L34)
[Only nodes_count can be updated](https://developer.openstack.org/api-ref/container-infrastructure-management/#update-information-of-cluster)

### CODE REVIEW

**Comment by ozerovandrei** [comment]
recheck

**Comment by liu-sheng** [comment]
recheck

File: openstack/resource_openstack_containerinfra_cluster_v1_test.go [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+  properties {
+    os_distro = "fedora-atomic"
+  }
+}
+
+resource "openstack_compute_keypair_v2" "keypair_1" {
+  name = "%s"
+}
+
+resource "openstack_compute_flavor_v2" "flavor_1" {
```
Comment (jtopjian) [inline openstack/resource_openstack_containerinfra_cluster_v1_test.go]: Let's have the flavor be defined with `OS_MAGNUM_FLAVOR` and `OS_MAGNUM_MASTER_FLAVOR`. This way, non-admin users can run the tests.

**Comment by mnaser** [comment]
@jtopjian I think this might be related to our new region, let me verify.

**Comment by mnaser** [comment]
@jtopjian please give it another shot, it *should* work now

**Comment by jtopjian** [comment]
@mnaser sorry for the late reply - busy day! Still seeing the same error.

**Comment by ozerovandrei** [comment]
@jtopjian Thank you for a thorough review!

**Comment by mnaser** [comment]
Better late than ever, but Magnum should be working now @jtopjian

**Comment by jtopjian** [comment]
@mnaser Nice! Thank you for following up!

**Pull request merged**